api/v1alpha2/awscluster_conversion.go

```diff
@@ -100,3 +102,26 @@ func Convert_v1alpha3_AWSClusterStatus_To_v1alpha2_AWSClusterStatus(in *infrav1a
 func Convert_v1alpha3_ClassicELB_To_v1alpha2_ClassicELB(in *infrav1alpha3.ClassicELB, out *ClassicELB, s apiconversion.Scope) error { //nolint
 	return autoConvert_v1alpha3_ClassicELB_To_v1alpha2_ClassicELB(in, out, s)
 }
+
+// Convert_v1alpha3_AWSLoadBalancerSpec_To_v1alpha2_AWSLoadBalancerSpec.
+func Convert_v1alpha3_AWSLoadBalancerSpec_To_v1alpha2_AWSLoadBalancerSpec(in *infrav1alpha3.AWSLoadBalancerSpec, out *AWSLoadBalancerSpec, s apiconversion.Scope) error { //nolint
+	if err := autoConvert_v1alpha3_AWSLoadBalancerSpec_To_v1alpha2_AWSLoadBalancerSpec(in, out, s); err != nil {
+		return err
+	}
+
+	if in.CrossZoneLoadBalancing != nil && *in.CrossZoneLoadBalancing {
+		return errors.New("cannot convert AWSLoadBalancerSpec crossZoneLoadBalancing field")
```

Comment:
Checked with apimachinery. We can't fail when downconverting because the older version doesn't have the newer field. We can do one of:

1. Silently drop the field
1. Add the new field to all API versions so that it's convertible (it would be present but unreconciled/unused in older versions)
1. Shuttle the field through metadata (annotation) so that a3 -> a2 -> a3 would preserve the value.

Reply:
Option 1 is probably a nonstarter because it would mean we'd lose data if we did a3 -> a2 -> update apiserver.